src/psyclone/dynamo0p3.py

```diff
@@ -2335,7 +2335,13 @@ def __init__(self, invoke, unique_psy_vars):
 
             # Create an object to capture info. on this inter-grid kernel
             # and store in our dictionary
-            self._ig_kernels[id(call)] = DynInterGrid(fine_arg, coarse_arg)
+            intergrid = DynInterGrid(fine_arg, coarse_arg)
+            # TODO #2503: Since we use the id(call) as a key, a copy of the
+            # call will not be able to use the ig_kernels to find its
+            # information, therefore we also store a direct link in the
+            # kernel
+            self._ig_kernels[id(call)] = intergrid
```

Comment:
Doing a quick grep, I see that (after your changes in this PR) we only ever loop over the `values` in the `_ig_kernels` dict and therefore it could perhaps just be a list instead. Perhaps add this as a task to #2503?

Reply:
Even better, I got rid of the cached dict/list altogether to avoid the double reference with self._schedule.walk(CodedKern)._itergrid_ref. Now the property looks for them dynamically, giving the right answer after tree schedule updates.

Reply:
Woohoo!